jerry-core/api/jerry.c

```diff
@@ -5383,12 +5383,9 @@ jerry_value_t
 jerry_get_resource_name (const jerry_value_t value) /**< jerry api value */
 {
 #if JERRY_RESOURCE_NAME
-  if (ecma_is_value_undefined (value))
+  if (ecma_is_value_undefined (value) && JERRY_CONTEXT (vm_top_context_p) != NULL)
```

Comment:
I would like to be this function as the `jerry_get_user_value` function below. It starts with:
```
  const ecma_compiled_code_t *bytecode_p = ecma_bytecode_get_from_value (value);
  if (bytecode_p == NULL)
  {
    return ECMA_VALUE_UNDEFINED;
  }
```

Then extract the string value. This will support more forms (e.g bound object, module) than the current one, you can even add some tests for the new forms.

This particular case can be kept though. Maybe this could be added to `jerry_get_user_value` for consistency (with tests)